test/python/compiler/test_disassembler.py

```diff
@@ -211,6 +224,163 @@ def test_multiple_conditionals_multiple_registers(self):
+
+        qobj = assemble(qc, FakeOpenPulse2Q())
+        dasm_circuits, _, _ = disassemble(qobj)
+
+        self.assertEqual(len(qc.calibrations), len(dasm_circuits[0].calibrations))
+        self.assertEqual(qc.calibrations.keys(), dasm_circuits[0].calibrations.keys())
+        self.assertEqual(all([
+            qc_cal.keys() == dasm_qc_cal.keys()
+            for qc_cal, dasm_qc_cal in
+            zip(qc.calibrations.values(), dasm_circuits[0].calibrations.values())]), True)
+        self.assertEqual(all([
+            _parametric_to_waveforms(qc_sched) == _parametric_to_waveforms(dasm_qc_sched)
+            for (_, qc_gate), (_, dasm_qc_gate) in
+            zip(qc.calibrations.items(), dasm_circuits[0].calibrations.items())
+            for qc_sched, dasm_qc_sched in zip(qc_gate.values(), dasm_qc_gate.values())]), True)
+
+    def test_parametric_pulse_circuit_calibrations(self):
+        """Test that disassembler parses parametric pulses back to pulse gates."""
+        with pulse.build() as h_sched:
+            pulse.play(pulse.library.Drag(50, 0.15, 4, 2), pulse.DriveChannel(0))
+
+        qc = QuantumCircuit(2)
+        qc.h(0)
+        qc.add_calibration('h', [0], h_sched)
+
+        backend = FakeOpenPulse2Q()
+        backend.configuration().parametric_pulses = ['drag']
+
+        qobj = assemble(qc, backend)
+        dasm_circuits, _, _ = disassemble(qobj)
+
+        self.assertEqual(len(qc.calibrations), len(dasm_circuits[0].calibrations))
+        self.assertEqual(qc.calibrations.keys(), dasm_circuits[0].calibrations.keys())
+        self.assertEqual(all([
+            qc_cal.keys() == dasm_qc_cal.keys()
+            for qc_cal, dasm_qc_cal in
+            zip(qc.calibrations.values(), dasm_circuits[0].calibrations.values())]), True)
+        self.assertEqual(all([
+            _parametric_to_waveforms(qc_sched) == _parametric_to_waveforms(dasm_qc_sched)
+            for (_, qc_gate), (_, dasm_qc_gate) in
+            zip(qc.calibrations.items(), dasm_circuits[0].calibrations.items())
+            for qc_sched, dasm_qc_sched in zip(qc_gate.values(), dasm_qc_gate.values())]), True)
+
+    def test_multi_circuit_uncommon_calibrations(self):
+        """Test that disassembler parses uncommon calibrations (stored at QOBJ experiment-level)."""
+        with pulse.build() as sched:
+            pulse.play(pulse.library.Drag(50, 0.15, 4, 2), pulse.DriveChannel(0))
+
+        qc_0 = QuantumCircuit(2)
+        qc_0.h(0)
+        qc_0.append(RXGate(np.pi), [1])
+        qc_0.add_calibration('h', [0], sched)
+        qc_0.add_calibration(RXGate(np.pi), [1], sched)
+
+        qc_1 = QuantumCircuit(2)
+        qc_1.h(0)
+
+        circuits = [qc_0, qc_1]
+        qobj = assemble(circuits, FakeOpenPulse2Q())
+        dasm_circuits, _, _ = disassemble(qobj)
+
+        self.assertEqual(
+            all([len(qc.calibrations) == len(dasm_qc.calibrations)
+                 for qc, dasm_qc in zip(circuits, dasm_circuits)]), True)
```

Comment:
If you add the suggested method, I believe (hope) that this section (L306-L322) would also be simplified:
```suggestion
         for qc, output_circ in zip(circuits, out_circuits):
             self.assertCircuitCalibrationsEqual(qc, output_circ)
```